test/functional/testlib.bash

```diff
@@ -1078,7 +1078,7 @@ remove_bundle() {
 	# may be used by another bundle)
 	dir_names=($(awk '/^D...\t/ { print $4 }' "$bundle_manifest"))
 	for dname in ${dir_names[@]}; do
-		sudo rmdir "$target_path$dname" 2> /dev/null
+		sudo rmdir --ignore-fail-on-non-empty "$target_path$dname" 2> /dev/null
```

Comment:
but this doesn't actually remove the directory? do we not care about that?

Reply:
Actually I think that is expected because it means the dir was not empty, so it may have at least one file from another test bundle that share that same directory, we don't want to delete files from another bundle.

The main use case I can think of this function is for doing partial clean ups when running many tests with the same test script, and all sharing the same test environment (using global_setup).

The whole test environment will be cleaned up after the tests in the script have finished, so there is no risk on leaving left overs.